src/cli/repl/server/connection.ts

```diff
@@ -277,6 +285,38 @@ export class FlowRServerConnection {
 		})
 	}
 
+	private handleLineageRequest(base: LineageRequestMessage) {
+		const requestResult = validateMessage(base, requestLineageMessage)
+
+		if(requestResult.type === 'error') {
+			answerForValidationError(this.socket, requestResult, base.id)
+			return
+		}
+
+		const request = requestResult.message
+		this.logger.info(`[${this.name}] Received lineage request for criterion ${request.criterion}`)
+
+		const fileInformation = this.fileMap.get(request.filetoken)
+		if(!fileInformation) {
+			sendMessage<FlowrErrorMessage>(this.socket, {
+				id:     request.id,
+				type:   'error',
+				fatal:  false,
+				reason: `The file token ${request.filetoken} has never been analyzed.`
+			})
+			return
+		}
+
+		const { dataflow: dfg, normalize: ast } = fileInformation.pipeline.getResults(true)
+		guard(dfg !== undefined, 'Dataflow graph must be present')
+		guard(ast !== undefined, 'AST must be present')
```

Comment:
Maybe add the filetoken to the error messages?
```suggestion
		guard(dfg !== undefined, `Dataflow graph must be present (request: ${request.filetoken})`)
		guard(ast !== undefined, `AST must be present (request: ${request.filetoken})`)
```